junitparser/xunit2.py

```diff
@@ -157,48 +156,71 @@ def system_err(self, value: str):
             self.append(err)
 
 
-class RerunFailure(RerunType):
+class RerunFailure(InterimResult):
     _tag = "rerunFailure"
 
 
-class RerunError(RerunType):
+class RerunError(InterimResult):
     _tag = "rerunError"
 
 
-class FlakyFailure(RerunType):
+class FlakyFailure(InterimResult):
     _tag = "flakyFailure"
 
 
-class FlakyError(RerunType):
+class FlakyError(InterimResult):
     _tag = "flakyError"
 
 
+RERUN_RESULTS = {RerunFailure, RerunError, FlakyFailure, FlakyError}
+
+
+R = TypeVar("R", bound=InterimResult)
+
+
 class TestCase(junitparser.TestCase):
     group = junitparser.Attr()
 
-    def _rerun_results(self, _type: T) -> List[T]:
-        elems = self.iterchildren(_type)
-        results = []
-        for elem in elems:
-            results.append(_type.fromelem(elem))
-        return results
+    # XUnit2 TestCase children are JUnit children and rerun results
+    ITER_TYPES = {t._tag: t for t in list(junitparser.TestCase.ITER_TYPES.values()) + list(RERUN_RESULTS)}
```

Comment:
This requires python 3.9, but is much nicer to read
```suggestion
    ITER_TYPES = junitparser.TestCase.ITER_TYPES | {t._tag: t for t in RERUN_RESULTS}
```

or `chain` the iterables, `itertools` is loaded allready so why not use it
```suggestion
    ITER_TYPES = {t._tag: t for t in itertools.chain(junitparser.TestCase.ITER_TYPES.values(), RERUN_RESULTS)}
```

or if we want to remove the global here as well:
```suggestion
    ITER_TYPES = {
        t._tag: t
        for t in itertools.chain(
            junitparser.TestCase.ITER_TYPES.values(),
            (RerunFailure, RerunError, FlakyFailure, FlakyError),
        )
    }
```